### PR TITLE
Detect the errors in format string for `log.InfoF` function and friends

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

This PR fixes several string format issues for `log.Infof` and friends.

Note: `%w` should be used when creating an error and not for formatting. `%w` is equivalent to `%v` when formatting an error.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/collector/corechecks/snmp/report/report_utils.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func processValueUsingSymbolConfig(value valuestore.ResultValue, symbol checkcon
 	if symbol.MatchPatternCompiled != nil {
 		strValue, err := value.ToString()
 		if err != nil {
-			log.Debugf("error converting value to string (value=%v):", value, err)
+			log.Debugf("error converting value to string (value=%v):%v", value, err)
```
Comment (vickenty) [inline pkg/collector/corechecks/snmp/report/report_utils.go:51]:
```suggestion
			log.Debugf("error converting value to string (value=%v): %v", value, err)
```

**Review by vickenty — APPROVED** [review]

File: tasks/go.py [hunk 2]
```diff
@@ -145,10 +145,15 @@ def vet(ctx, targets, rtloader_root=None, build_tags=None, arch="x64"):
...
+    printf_funcs = "-printf.funcs"
+    for fct in ["Tracef", "Debugf", "Infof", "Printf", "Warnf", "Errorf"]:
+        printf_funcs += printf_funcs_sep + "github.com/DataDog/datadog-agent/pkg/util/log." + fct
+        printf_funcs_sep = ","
+
     _, _, env = get_build_flags(ctx, rtloader_root=rtloader_root)
     env["CGO_ENABLED"] = "1"
-
-    ctx.run(f"go vet -tags \"{' '.join(tags)}\" " + " ".join(args), env=env)
+    ctx.run(f"go vet {printf_funcs} -tags \"{' '.join(tags)}\" " + " ".join(args), env=env)
```
Comment (djmitche) [inline tasks/go.py:156]: Nice!

**Review by djmitche — APPROVED** [review]
Looks good.  Especially for Errorf, which in fact returns an `error`, I can see why folks are tempted to use `%w`.  I recall looking at this a while back, and it wasn't clear how to support `%w` in that context, so I think prohibiting it makes sense.

**Review by just-chillin — APPROVED** [review]
Looked at the changes in the `processes` package. Looks great! Thank you very much!

**Review by ahmed-mez — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]
Looks good for `pkg/workloadmeta/collectors/cloudfoundry/cloudfoundry.go`

**Review by gmarabout — APPROVED** [review]
Reviewed only the snmp part. Looks good to me.

File: pkg/collector/corechecks/snmp/checkconfig/config_metric.go [hunk 3]
```diff
@@ -178,7 +178,7 @@ func (mtc *MetricTagConfig) GetTags(value string) []string {
 				normalizedTemplate := normalizeRegexReplaceValue(val)
 				replacedVal := RegexReplaceValue(value, mtc.pattern, normalizedTemplate)
 				if replacedVal == "" {
-					log.Debugf("pattern `%v` failed to match `%v` with template `%v`", value, normalizedTemplate)
+					log.Debugf("pattern `%v` failed to match `%v` with template `%v`", value, mtc.pattern, normalizedTemplate)
```
Comment (AlexandreYang) [inline pkg/collector/corechecks/snmp/checkconfig/config_metric.go:181]:
```suggestion
					log.Debugf("pattern `%v` failed to match `%v` with template `%v`", mtc.pattern, value, normalizedTemplate)
```

**Review by AlexandreYang — COMMENTED** [review]
Nice, thanks. Just have a minor comment below:

File: tasks/go.py (hunk 2 above)
Comment (AlexandreYang) [reply tasks/go.py:156]: 👏

**Review by mx-psi — CHANGES_REQUESTED** [review]
Tests are failing with 

```
# github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/cri
pkg/collector/corechecks/containers/cri/check.go:78:3: Warnf call has error-wrapping directive %w, which is only supported by Errorf
# github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/containerd
pkg/collector/corechecks/containers/containerd/check.go:81:3: Warnf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/containerd/check_metrics_extension.go:36:3: Debugf call has error-wrapping directive %w, which is only supported by Errorf
# github.com/DataDog/datadog-agent/pkg/collector/corechecks/containers/docker
pkg/collector/corechecks/containers/docker/check.go:81:3: Warnf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/docker/check.go:86:3: Warnf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/docker/check.go:199:5: Warnf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/docker/check.go:224:4: Warnf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/docker/check.go:274:3: Warnf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/docker/check.go:275:7: Warnf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/docker/check.go:281:3: Warnf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/docker/check.go:282:7: Warnf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/docker/check.go:292:5: Tracef format %w reads arg #2, but call has 1 arg
pkg/collector/corechecks/containers/docker/check_metrics_extension.go:36:3: Debugf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/docker/check_network_linux.go:73:3: Debugf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/docker/check_network_linux.go:79:3: Debugf call has error-wrapping directive %w, which is only supported by Errorf
pkg/collector/corechecks/containers/docker/check_network_linux.go:193:3: Debugf format %s reads arg #1, but call has 0 args
# github.com/DataDog/datadog-agent/pkg/workloadmeta/collectors/docker
pkg/workloadmeta/collectors/docker/docker.go:69:3: Warnf call has error-wrapping directive %w, which is only supported by Errorf
pkg/workloadmeta/collectors/docker/docker.go:240:5: Debugf call has error-wrapping directive %w, which is only supported by Errorf
# github.com/DataDog/datadog-agent/pkg/logs/internal/launchers/docker
pkg/logs/internal/launchers/docker/launcher.go:83:13: Errorf call has arguments but no formatting directives
# github.com/DataDog/datadog-agent/pkg/util/containers/v2/metrics/system
pkg/util/containers/v2/metrics/system/collector_disk_linux.go:44:3: Debugf call has error-wrapping directive %w, which is only supported by Errorf
pkg/util/containers/v2/metrics/system/collector_disk_linux.go:106:3: Debugf call has error-wrapping directive %w, which is only supported by Errorf
pkg/util/containers/v2/metrics/system/collector_network_linux.go:34:4: Debugf call has error-wrapping directive %w, which is only supported by Errorf
# github.com/DataDog/datadog-agent/pkg/util/containers/v2/metrics/docker
pkg/util/containers/v2/metrics/docker/collector_linux.go:83:3: Debugf call has error-wrapping directive %w, which is only supported by Errorf
# github.com/DataDog/datadog-agent/pkg/util/ecs/metadata
pkg/util/ecs/metadata/clients.go:88:3: Debugf call has error-wrapping directive %w, which is only supported by Errorf
# github.com/DataDog/datadog-agent/pkg/util/kernel
pkg/util/kernel/find_headers.go:87:3: Debugf call has error-wrapping directive %w, which is only supported by Errorf
pkg/util/kernel/find_headers.go:142:4: Debugf call has error-wrapping directive %w, which is only supported by Errorf
```

File: pkg/collector/corechecks/snmp/checkconfig/config_metric.go (hunk 3 above)
Comment (pducolin) [reply pkg/collector/corechecks/snmp/checkconfig/config_metric.go:181]: **❓ question**
Should this be 
```suggestion
					log.Debugf("value `%v` failed to match pattern `%v` with template `%v`", value, mtc.pattern, normalizedTemplate)
```
?

**Review by xlucas — APPROVED** [review]

**Review by AlexandreYang — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

**Review by coignetp — APPROVED** [review]
👍  for remote-config file

**Review by mx-psi — APPROVED** [review]

**Pull request merged**